meshery.sh

```diff
@@ -36,7 +36,10 @@ main() {
 	mv ~/minified_config ~/.kube/config
 
 	curl -L https://git.io/meshery | DEPLOY_MESHERY=false bash -
-	#mesheryctl system context create new-context --adapters $service_mesh_adapter --platform docker --url http://localhost:9081 --set --yes
+
+	if [ ! -z "$service_mesh_adapter" ]; then
```

Comment:
@suhaniii checking if the variable is empty is a great consideration :+1:.
But I think we're pretty sure it won't be empty.

Check one: `service_mesh` is a required input by the github action which means this won't run if it is empty.

Check two: When the command line flags provided while calling `meshery.sh` are parsed. We check if the flag argument for `--service-mesh` is empty, if it is, we exit with an error and if not only then we create the `service_mesh_adapter` variable.

https://github.com/layer5io/meshery-smi-conformance-action/blob/56f3439683b4c02af2693060a86e5ea4a492daaa/meshery.sh#L82-L90

Reply:
Hmm, if that is the case, I am not sure what other changes are made here that would fix the issue that we are having.

Reply:
Adding if clause solves the problem describe in issue. CI which you are referring to, is that integrated here?

Reply:
@DelusionalOptimist   Reg 
_Check one: service_mesh is a required input by the github action which means this won't run if it is empty._.  
Can we add else part as well, so that it will run when its empty?

Reply:
@DelusionalOptimist  Can you try these changes and let me know the issue?

Reply:
According to me id `service_mesh` is empty then it will exit with error, if not empty `service_mesh_adapter` got created.  Then under if clause `mesheryctl` is able to parse `service_mesh_adapter`

Reply:
Hmm... :thinking:. Sure @suhaniii. Testing this :+1:

Reply:
Do you also happen to have logs of any runs of this with your changes? If yes, please provide them as well : )

Reply:
My main concern was mesheryctl able to parse the var

1. Working one
```
suhaniagarwal@Suhanis-MacBook-Air osrc % cat abc.sh
#!/usr/bin/env bash
service_mesh_adapter="meshery-linkerd"
echo $service_mesh_adapter

if [ ! -z "$service_mesh_adapter" ]; then
   mesheryctl system context create new-context --adapters $service_mesh_adapter --platform docker --url http://localhost:9081 --set --yes
fi
```
```
suhaniagarwal@Suhanis-MacBook-Air osrc % sh -x abc.sh
+ service_mesh_adapter=meshery-linkerd
+ echo meshery-linkerd
meshery-linkerd
+ '[' '!' -z meshery-linkerd ']'
+ mesheryctl system context create new-context --adapters meshery-linkerd --platform docker --url http://localhost:9081 --set --yes
ERRO[0000] failure getting variant                       error="getCPUInfo for OS darwin: not implemented"
Added `new-context` context
```

2. Not working
```
suhaniagarwal@Suhanis-MacBook-Air osrc % cat abc.sh
#!/usr/bin/env bash
service_mesh_adapter="meshery-linkerd"
echo $service_mesh_adapter

mesheryctl system context create new-context --adapters $service_mesh_adapter --platform docker --url http://localhost:9081 --set --yes
```
```
suhaniagarwal@Suhanis-MacBook-Air osrc % sh -x abc.sh
+ service_mesh_adapter=meshery-linkerd
+ echo meshery-linkerd
meshery-linkerd
+ mesheryctl system context create new-context --adapters meshery-linkerd --platform docker --url http://localhost:9081 --set --yes
ERRO[0000] failure getting variant                       error="getCPUInfo for OS darwin: not implemented"
Error: error adding context: a context with same name already exists
Usage:
  mesheryctl system context create <context-name> [flags]

Flags:
  -a, --adapters stringArray   List of adapters
  -h, --help                   help for create
  -p, --platform string        Platform to deploy Meshery
  -s, --set                    Set as current context
  -u, --url string             Meshery Server URL with Port

Global Flags:
      --config string    config file (default location is: /Users/suhaniagarwal/.meshery/config.yaml) (default "/Users/suhaniagarwal/.meshery/config.yaml")
  -c, --context string   (optional) temporarily change the current context.
  -v, --verbose          verbose output
  -y, --yes              (optional) assume yes for user interactive prompts.

suhaniagarwal@Suhanis-MacBook-Air osrc %
```

Reply:
I might be wrong. And now I don't get the problem 🤔

Reply:
Yeah this has been a bit of a head-scratcher :sweat_smile:. Our main trouble is that the currently used command is working fine in our local systems but not on the CI runner.

Reply:
May I ping you on slack @suhaniii?